### ISSUE TITLE
Is it possible to use library without WinRt?

### ISSUE THREAD

**Issue by alex-aparin** [axodox/axodox-machinelearning#9]
Recently I tried to use library in my console application. It is important to note that application is not UWP based. Unfortunately I encountered error - `Exception thrown at 0x00007FFD7857CF19 (KernelBase.dll) in Console.exe: WinRT originate error - 0x80073D54 : 'The process has no package identity.'.` It seems like that WinRt restricts model of application. Is there any way to exclude usage of Winrt? or another way to combine WinRt library with non uwp application?

**Comment by axodox** [comment]
Can you send the stack trace?

WinRT APIs do not require a package identity, however I might have made a mistake.

**Comment by alex-aparin** [comment]
It originates when I am trying to create OnnxEnvironment. 
![image](https://github.com/axodox/axodox-machinelearning/assets/20486820/9429c2ec-8e41-49df-8555-bc1c9333d20d)

**Comment by axodox** [comment]
Oh, well... I messed it up with the WinRT logging. Please commit out the logger in the header and where it is used, I will push a fix for it later.

**Comment by alex-aparin** [comment]
Thank you very much! It's awesome how you instantly replies on issue

**Comment by axodox** [comment]
FYI this is the issue: https://github.com/axodox/axodox-common/blob/main/Axodox.Common.Shared/Infrastructure/Logger.cpp
We will need to surround this part with an `#ifdef`:
![image](https://github.com/axodox/axodox-machinelearning/assets/22882208/2ac834f0-1df7-4a29-8a69-7087d847dee8)

**Comment by axodox** [comment]
I will need to read up on this, it seems that the logging API requires package identity. To clarify package identity != UWP, though all UWP apps have package identity. Also WinRT != UWP.

UWP is application model which runs the app in an isolated container.
WinRT is a new familily of Windows APIs it is not tied to UWP, it works in any app.
Package identity is tied to MSIX packages, it is not tied to UWP or WinRT.

There are some APIs which require a package identity. There are APIs to check if the app has package identity, so probably it won't be an `#ifdef`, just an if with a check for package identity. I think there is also some API to add package identity to apps which do not have it, but I never tried that in action.

**Comment by axodox** [comment]
So no the includes are fine, I have used WinRT in a normal unreal engine app (so not UWP, no MSIX), and had no issue.

**Comment by alex-aparin** [comment]
Thank you, I grasped the idea. It will help me hugely to diagnose next possible errors

**Comment by axodox** [comment]
One thing to look out for, that you will need to reference the nuget packages from DirectML and ONNX runtime, as well as have the model files, and also the clip_tok.onnx and ortextensions.dll from my nuget.

I plan to refine the nuget later to require these dependencies and copy the mentioned files automatically.

**Comment by alex-aparin** [comment]
@axodox yep. I have ran into that dependencies. But I temporary included my project within axodox-machinelearning solution. Later I will refactor it
